src/components/Operations.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import { BrowserRouter as Router, Route, Link, Redirect} from 'react-router-dom'
+import Transactions from './Transactions'
+import axios from 'axios'
+
+class Operations extends Component {
+
+  
+    constructor() {
+        super()
+        this.state = {
+            amount: "",
+            vendor: "",
+            category: ""
+        }
+    }
+
+    
+    handleAmount = event => {
+        this.setState({amount: event.target.value});
+    }
+
+    handleVendor = event => {
+        this.setState({vendor: event.target.value});
+    }
+
+    handleCategory = event => {
+        this.setState({category: event.target.value});
+    }
+
+    getBalance = () =>{
+        return this.props.balance()
+    }
+
+    deposit = () => {
+        if(this.state.amount.length > 0 && this.state.vendor.length > 0 && this.state.category.length > 0) {
+            
+            axios({
+                method: 'post',
+                url: 'http://localhost:2700/transaction',
+                headers: {}, 
+                data: { 
+                    amount: 100,
+                    vendor: 'rami',
+                    category: 'foods' }
+            })
+            this.props.deposit(this.state.amount, this.state.vendor, this.state.category)
+            this.setState({
+                amount: "",
+                vendor: "",
+                category: ""
+            });
+        }
```

Comment:
It's a promise! you should handle local changes after resolving the promise (finishing the POST request)